src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2232,6 +2232,12 @@ internal void MarkStaticConstructorVisibleToReflection(TypeDefinition type, in D
                         // For methods that must be preserved, blame the declaring type.
                         MarkMethod(method, new DependencyInfo(DependencyKind.VirtualNeededDueToPreservedScope, type), typeOrigin);
                     }
+                    else if (method.IsExtensionMarkerMethod())
+                    {
+                        // Extension marker methods are private, but are key metadata used by tooling,
+                        // they should be preserved if the corresponding type is preserved.
+                        MarkMethod(method, new DependencyInfo(DependencyKind.TypePreserve, type), typeOrigin);
+                    }
```

Comment:
CC. @dotnet/illink 

Could I get review on this and input on if there is a better way to handle this. The general issue is that `extensions members` end up adding a private "extension marker method" which impacts various tools and how they observe the IL.

In particular, it is impactful to tools like the ApiCompat analyzer.

Reply:
It does not sound right for the extension markers to be private if they impact public surface. Should this be fixed in Roslyn instead?

Reply:
👍, that definitely sounds reasonable. I would defer to the Roslyn team on if that is appropriate or if this is a case where tools need to understand certain private metadata (as the linker already has to do with certain other private `unspeakable` types and methods generated).